### PR TITLE
Feat/pro 926/device create event

### PR DESCRIPTION
**Pull request by CalumMcCall**

### CODE REVIEW

File: common-messaging/src/main/scala/org/genivi/sota/messaging/MessageBusManager.scala [hunk 1]
```diff
@@ -7,30 +7,50 @@ import akka.actor.ActorSystem
...
 import com.typesafe.config.{Config, ConfigException}
 import io.circe.Decoder
 import io.circe.parser._
-import org.genivi.sota.messaging.Messages.DeviceSeenMessage
+import org.genivi.sota.data.Device.DeviceName
+import org.genivi.sota.messaging.Messages.{DeviceCreatedMessage, DeviceSeenMessage}
 import org.genivi.sota.messaging.kinesis.KinesisClient
 import org.genivi.sota.messaging.nats.NatsClient
 
 object Messages {
```
Comment (koshelev) [inline common-messaging/src/main/scala/org/genivi/sota/messaging/MessageBusManager.scala]: I think we should move messages in a separate file

File: common-messaging/src/main/scala/org/genivi/sota/messaging/kinesis/KinesisClient.scala [hunk 2]
```diff
@@ -51,32 +51,46 @@ object KinesisClient {
...
+
+  def getDeviceSeenPublisher(system: ActorSystem,
+                             config: Config): ConfigException Xor (DeviceSeenMessage => Unit) =
+    getAmazonClient(system, config).map { client =>
       (msg: DeviceSeenMessage) =>
         {
-          import io.circe.syntax._
-          import org.genivi.sota.marshalling.CirceInstances._
-          client.putRecord(streamName, ByteBuffer.wrap(msg.asJson.noSpaces.getBytes), msg.deviceId.underlying.get)
+          client.putRecord(DeviceSeenMessage.getClass.getName, ByteBuffer.wrap(msg.asJson.noSpaces.getBytes),
```
Comment (koshelev) [inline common-messaging/src/main/scala/org/genivi/sota/messaging/kinesis/KinesisClient.scala]: If you are using a class as a partition key then you are actually prevents kinesis from sharding the messages

Comment (CalumMcCall) [inline common-messaging/src/main/scala/org/genivi/sota/messaging/kinesis/KinesisClient.scala]: The classname is used for the stream name. The partition key is the deviceId.

Comment (koshelev) [inline common-messaging/src/main/scala/org/genivi/sota/messaging/kinesis/KinesisClient.scala]: Oh, sorry, my fault

**Comment by koshelev** [comment]
Is it still WIP?

**Comment by CalumMcCall** [comment]
Whoops, forgot to remove the WIP. It's ready to merge now.

**Pull request merged**